React/Modules/RCTDevSettings.m

```diff
@@ -0,0 +1,508 @@
+  }
+  [_keysChanged addObject:key];
+  [[NSUserDefaults standardUserDefaults] setObject:_settings forKey:kRCTDevSettingsUserDefaultsKey];
+  [[NSUserDefaults standardUserDefaults] synchronize];
+}
+
+- (id)storedSettingForKey:(NSString *)key
+{
+  return _settings[key];
+}
+
+- (void)_userDefaultsDidChange
+{
+  NSDictionary *userInfo;
+  if (_keysChanged) {
+    userInfo = @{ kRCTDevSettingsUpdatedSettingsKey: [_keysChanged allObjects] };
+    _keysChanged = nil;
+  }
+  _settings = [NSMutableDictionary dictionaryWithDictionary:[[NSUserDefaults standardUserDefaults] objectForKey:kRCTDevSettingsUserDefaultsKey]];
+  [[NSNotificationCenter defaultCenter] postNotificationName:kRCTDevSettingsDidUpdateNotification object:self userInfo:userInfo];
+}
+
+- (void)_reloadWithDefaults:(NSDictionary *)defaultValues
+{
+  NSDictionary *existingSettings = [[NSUserDefaults standardUserDefaults] objectForKey:kRCTDevSettingsUserDefaultsKey];
+  _settings = (existingSettings) ? [existingSettings mutableCopy] : [NSMutableDictionary dictionary];
+  for (NSString *key in [defaultValues keyEnumerator]) {
+    if (!_settings[key]) {
+      _settings[key] = defaultValues[key];
+    }
+  }
+  [[NSUserDefaults standardUserDefaults] setObject:_settings forKey:kRCTDevSettingsUserDefaultsKey];
+  [[NSUserDefaults standardUserDefaults] synchronize];
+}
+
+@end
+
+@interface RCTDevSettings () <RCTBridgeModule, RCTInvalidating, RCTWebSocketObserverDelegate>
+{
+  NSURLSessionDataTask *_liveReloadUpdateTask;
+  NSURL *_liveReloadURL;
+  BOOL _isJSLoaded;
+  NSString *_executorOverride;
+}
+
+@property (nonatomic, strong) Class executorClass;
+@property (nonatomic, readwrite, strong) id<RCTDevSettingsDataSource> dataSource;
+
+@end
+
+@implementation RCTDevSettings
+
+@synthesize bridge = _bridge;
+
+RCT_EXPORT_MODULE()
+
+- (instancetype)init
+{
+  // default behavior is to use NSUserDefaults
+  NSDictionary *defaultValues = @{
+    kRCTDevSettingShakeToShowDevMenu: @YES,
+  };
+  RCTDevSettingsUserDefaultsDataSource *dataSource = [[RCTDevSettingsUserDefaultsDataSource alloc] initWithDefaultValues:defaultValues];
+  return [self initWithDataSource:dataSource];
+}
+
+- (instancetype)initWithDataSource:(id<RCTDevSettingsDataSource>)dataSource
+{
+  if (self = [super init]) {
+    _dataSource = dataSource;
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(jsLoaded:)
+                                                 name:RCTJavaScriptDidLoadNotification
+                                               object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(_dataSourceDidUpdate:)
+                                                 name:kRCTDevSettingsDidUpdateNotification
+                                               object:_dataSource];
+    
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
```

Comment:
Good catch, it was just copied over from the old implementation and I overlooked it.